WMF Framework/SharedContainerCache.swift

```diff
@@ -25,20 +28,79 @@ public final class SharedContainerCache<T: Codable> {
         return cacheDirectoryContainerURL.appendingPathComponent(pathComponent.rawValue).appendingPathExtension("json")
     }
     
+    private func subdirectoryFileURL(to subfolder: String) -> URL {
+        return cacheDirectoryContainerURL
+            .appendingPathComponent(pathComponent.rawValue, isDirectory: true)
+    }
+
+    func cacheFileURL(to subfolder: String) -> URL {
+        return cacheDirectoryContainerURL
+            .appendingPathComponent(pathComponent.rawValue, isDirectory: true)
+            .appendingPathComponent(subfolder)
+            .appendingPathExtension("json")
+    }
+    
     public func loadCache() -> T {
-        if let data = try? Data(contentsOf: cacheDataFileURL), let decodedCache = try? JSONDecoder().decode(T.self, from: data) {
+        let filePath = fileFolder.isEmpty ? cacheDataFileURL : cacheFileURL(to: fileFolder)
+        if let data = try? Data(contentsOf: filePath), let decodedCache = try? JSONDecoder().decode(T.self, from: data) {
             return decodedCache
         }
 
         return defaultCache()
     }
 
     public func saveCache(_ cache: T) {
+        if fileFolder.isEmpty {
```

Comment:
This method could go back to how it was before. But before writing, create the subdirectory if needed:

```
public func saveCache(_ cache: T) {

        let encoder = JSONEncoder()
        guard let encodedCache = try? encoder.encode(cache) else {
            return
        }

        if let subdirectoryURL = subdirectoryURL {
            try? FileManager.default.createDirectory(at: subdirectoryURL, withIntermediateDirectories: true, attributes: nil)
        }

        try? encodedCache.write(to: cacheDataFileURL)
```